### PR TITLE
CRD Controller

### PR DESCRIPTION
**Pull request by lkysow**
Changes proposed in this PR:
- Add new commands `controller` and `webhook-cert-manager` 
- Add operator-sdk files needed to generate CRDs and controller code

Checklist:
- [X] Tests added

=> changelog will be added in upcoming PR

- [ ] CHANGELOG entry added (*HashiCorp engineers only, community PRs should not add a changelog entry*)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,13 +1,54 @@
...
 executors:
   go:
     docker:
       - image: circleci/golang:1.14
     environment:
       - TEST_RESULTS: /tmp/test-results # path to where test results are saved
-      - CONSUL_VERSION: 1.8.2 # Consul's OSS version to use in tests
-      - CONSUL_ENT_VERSION: 1.8.2+ent # Consul's enterprise version to use in tests
+      - CONSUL_VERSION: aa0f5ff839c515aad3baa38c7936b4630263ca89 # Consul's OSS version to use in tests
+      - CONSUL_ENT_VERSION: 511f5942610bfa3ae53a40ca05db1858b25c2263 # Consul's enterprise version to use in tests
```
Comment (lkysow) [inline .circleci/config.yml:11]: Todo: we need these to point at 1.9.0 so the controller tests pass.

File: catalog/to-consul/resource.go [hunk 2]
```diff
@@ -357,19 +358,14 @@ func (t *ServiceResource) generateRegistrations(key string) {
 	}
 
 	// Update the Consul namespace based on namespace settings
```
Comment (lkysow) [inline catalog/to-consul/resource.go:360]: Double check this

**Review by ashwin-venkatesh — APPROVED** [review]
🦊 🐧  This is all approved so who am i to get in the way

**Review by kschoche — APPROVED** [review]
Excellent work!

**Pull request merged**